controllers/ctltools.js

```diff
@@ -77,12 +77,27 @@ const getUpdateHandler = (req, res, next) => {
     });
 };
 
+const convertZerosToString = (obj) => {
+    Object.keys(obj).forEach(key => {
+        if (typeof obj[key] === 'object') {
+            convertZerosToString(obj[key]);
+        } else if (obj[key] === 0) {
+            obj[key] = '0';
+        }
+    });
+}
+
 const getDetailHandler = (req, res, next) => {
-    api(req, { version: 'v3' }).get(`/tools/external-tools/${req.params.id}`).then(data => {
-        if (data.config.type === 'oauth2') {
-            data.config.redirectUris = data.config.redirectUris.join(';');
+    Promise.all([
+        api(req, { version: 'v3' }).get(`/tools/external-tools/${req.params.id}`),
+        api(req, { version: 'v3' }).get(`/tools/external-tools/${req.params.id}/metadata`)
+    ]).then(([toolData, toolMetaData]) => {
+        if (toolData.config.type === 'oauth2') {
+            toolData.config.redirectUris = toolData.config.redirectUris.join(';');
         }
-        res.json(data);
+
+        convertZerosToString(toolMetaData);
+        res.json({...toolData, ...toolMetaData});
     }).catch(err => {
         next(err);
```

Comment:
In case of an error we will see the general error page.. also on loading metadata. Is it wanted like that?

Reply:
yes I think so, there were no specific requirements for it